Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -544,6 +544,25 @@ private DeployResult GetResult(string id, string activeDeploymentId, bool isDepl
                 string repositoryRoot = _environment.RepositoryPath;
                 var perDeploymentSettings = DeploymentSettingsManager.BuildPerDeploymentSettingsManager(repositoryRoot, _settings);
 
+                string delayMaxInStr = perDeploymentSettings.GetValue(SettingsKeys.MaxRandomDelayInSec, false);
+                if (delayMaxInStr != null) {
+                    int maxDelay;
+                    if(!Int32.TryParse(delayMaxInStr, out maxDelay) || maxDelay < 0)
+                    {
+                        tracer.Trace("Invalid {0} value, expect a positive integer, received {1}", SettingsKeys.MaxRandomDelayInSec, delayMaxInStr);
+                    }else
+                    {
+                        tracer.Trace("{0} is set to {1}s", SettingsKeys.MaxRandomDelayInSec, maxDelay);
+                        Random rdn = new Random();
+                        int gap = rdn.Next(maxDelay);
+                        using (tracer.Step("Randomization applied to {0}, Start sleeping for {1}s", maxDelay,gap))
```

Comment:
Here we also need to log something to the user visible deployment log. `tracer` is a separate thing that is more for internals, and customer doesn't normally see it. You probably just need to call `innerLogger.Log("...")`.